Packs/ContentTesting/Scripts/UnitTestPlaybookAnalyzer/UnitTestPlaybookAnalyzer.py

```diff
@@ -31,181 +31,125 @@ class TaskStat(TypedDict):
 
     return tasks
 
 
-def TaskStats(task: list[Task], taskstat: dict[str, TaskStat]) -> dict[str, TaskStat]:
+def TaskStats(task: List[Task], taskstat: Dict[str, TaskStat]) -> Dict[str, TaskStat]:
     for t in task:
-        tid = t['tid']
+        taskid = t['tid']
         dur = t['duration']
-        if tid not in taskstat:
-            taskstat[tid] = {'tid': tid, 'name': t['name'], 'mindur': 0, 'maxdur': 0, 'avgdur': 0, 'totdur': 0,
-                             'count': 0, 'completed': 0, 'started': 0, 'notexecuted': 0, 'error': 0, 'waiting': 0}  # type: ignore
+        if taskid not in taskstat:
+            taskstat[taskid] = TaskStat(tid=taskid, name=t['name'], mindur=None, maxdur=0, avgdur=0,
+                                        totdur=0, count=0, completed=0, started=0, notexecuted=0, error=0, waiting=0)
         if t['state'] == "Completed":
-            if dur > taskstat[tid]['maxdur']:  # type: ignore
-                taskstat[tid]['maxdur'] = dur  # type: ignore
-            if taskstat[tid]['mindur'] is not None:  # type: ignore
-                if dur < taskstat[tid]['mindur']:  # type: ignore
-                    taskstat[tid]['mindur'] = dur  # type: ignore
+            if dur > taskstat[taskid]['maxdur']:
+                taskstat[taskid]['maxdur'] = dur
+            if taskstat[taskid]['mindur'] != None:
+                if dur < taskstat[taskid]['mindur']:
+                    taskstat[taskid]['mindur'] = dur
             else:
-                taskstat[tid]['mindur'] = dur  # type: ignore
-            taskstat[tid]['totdur'] += dur  # type: ignore
-            taskstat[tid]['completed'] += 1  # type: ignore
+                taskstat[taskid]['mindur'] = dur
+            taskstat[taskid]['totdur'] += dur
+            taskstat[taskid]['completed'] += 1
         elif t['state'] == "Error":
-            taskstat[tid]['error'] += 1  # type: ignore
+            taskstat[taskid]['error'] += 1
         elif t['state'] == "Waiting":
-            taskstat[tid]['waiting'] += 1  # type: ignore
+            taskstat[taskid]['waiting'] += 1
         else:
-            taskstat[tid]['started'] += t['started']  # type: ignore
-            taskstat[tid]['notexecuted'] += t['notexecuted']  # type: ignore
-        taskstat[tid]['count'] += 1  # type: ignore
+            taskstat[taskid]['started'] += t['started']
+            taskstat[taskid]['notexecuted'] += t['notexecuted']
+        taskstat[taskid]['count'] += 1
 
     for key, ts in taskstat.items():
         ts['avgdur'] = int(ts['totdur'] / ts['count'])
 
     return taskstat
 
 
-def GetTaskStats(playbookname: str, occurred: str) -> tuple[dict[str, TaskStat], int]:
-    argument = {'query': f'playbook:"{playbookname}" occurred:>="{occurred}"', 'size': 1000}
+def GetTaskStats(playbookname, subplaybookname, firstday, lastday, maxinc):
+    argument = {'query': f'playbook:"{playbookname}" occurred:>="{firstday}T00:00:00" and occurred:<="{lastday}T23:59:59"'}
```

Comment:
The arguments here aren't required (aren't mandatory), thus can be empty. It is recommended to add to the YML of the script 'required: true'.